tf2_ros/src/buffer.cpp

```diff
@@ -68,6 +68,10 @@ Buffer::Buffer(rclcpp::Clock::SharedPtr clock, tf2::Duration cache_time) :
 
   jump_handler_ = clock_->create_jump_callback(nullptr, post_jump_cb, jump_threshold);
 
+  if (node_) {
+    frames_server_ = node_->create_service<tf2_msgs::srv::FrameGraph>(
```

Comment:
The TODO block below is equivalent to this except that this is missing the logic not to doubly create a service if the node already has this service. We should integrate that and then the TODO can go away.

Reply:
Can you please apply the suggested change to the code.

Reply:
@tfoote Any update on this?

Reply:
@tfoote @dirk-thomas , any update?

> is missing the logic not to doubly create a service if the node already has this service

https://github.com/ros2/geometry2/blob/d142c9efb6f51d60105c4c127178b31bf291f2eb/tf2_ros/src/buffer.cpp#L75-L80

 I'm not sure what @tfoote is suggesting; by "node", do you mean any node in the graph, or just the node from the node pointer. When would this happen, when the user creates two tf buffers in one node?

Reply:
I don't think there's much we can do about checking for duplicate service creation, since a remote node could always create a service with the same name. We could probably guard against the local node from creating the same service twice, but I don't know the cases where this can happen and what the consequences are.

Reply:
> I don't think there's much we can do about checking for duplicate service creation, since a remote node could always create a service with the same name.

Agreed, that is why I think this change should be merged as is.

> We could probably guard against the local node from creating the same service twice, but I don't know the cases where this can happen and what the consequences are.

Since the same applies to pretty much any existing code creating a service I don't see why in this case we should add any kind of protection (which doesn't existing any all the other cases).

Reply:
I'd say we can remove the TODO below; if there's an issue with not adding the guard, we can always add it retroactively.

Reply:
Done in e245141130405a558d34394d6f4da7bdfd0f6933.